pylsp/python_lsp.py

```diff
@@ -358,7 +360,7 @@ def execute_command(self, command, arguments):
         return self._hook('pylsp_execute_command', command=command, arguments=arguments)
 
     def format_document(self, doc_uri, options):
-        return self._hook('pylsp_format_document', doc_uri, options=options)
+        return lambda: self._hook('pylsp_format_document', doc_uri, options=options)
```

Comment:
What are the advantages of making this call async?

Reply:
it's possible to keep progress reporting for it.

Reply:
Ok, I understand now: you removed it for `format_range` but leave it for `format_document`. That's fine by me.